pkg/asset/installconfig/aws/availabilityzones.go

```diff
@@ -29,19 +28,9 @@ func availabilityZones(ctx context.Context, session *session.Session, region str
 		return nil, errors.Wrap(err, "fetching availability zones")
 	}
 
-	// This finds local zones and allows us to filter them out. It does so by
-	// using the fact that local zones have a predictable format based on the
-	// region, for example a local zone in us-west-2 will look like
-	// "us-west-2-las-1a". This is calculated relative to the region name as not
-	// all regions follow the same format with 2 hyphens.
-	// This is a temporary fix until we update our aws-sdk-go package to a
-	// version that adds the ZoneType field to the AvailabilityZone struct
-	numberOfHyphensInRegion := strings.Count(region, "-")
-
 	zones := []string{}
 	for _, zone := range resp.AvailabilityZones {
-		numberOfHyphensInZone := strings.Count(*zone.ZoneName, "-")
-		if numberOfHyphensInZone != numberOfHyphensInRegion+2 {
+		if *zone.ZoneType != "local-zone" {
```

Comment:
I suggest that we enforce that the zone *must be* an availability zone rather than simply not a local zone.
```suggestion
		if *zone.ZoneType == "availability-zone" {
```